src/main/java/org/broadinstitute/dsm/route/BatchKitsRoute.java

```diff
@@ -0,0 +1,55 @@
+package org.broadinstitute.dsm.route;
+
+import com.google.gson.Gson;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.KitRequestShipping;
+import org.broadinstitute.dsm.statics.RequestParameter;
+import org.broadinstitute.dsm.util.SystemUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BatchKitsRoute implements Route {
+    private static final Logger logger = LoggerFactory.getLogger(BatchKitsRoute.class);
+
+    @Override
+    public Object handle(Request request, Response response) throws Exception {
+        String study = request.params(RequestParameter.REALM);
+        String ids = null;
+        if (StringUtils.isNotBlank(study)) {
+            HttpServletRequest rawRequest = request.raw();
+            String content = SystemUtil.getBody(rawRequest);
+            ids = (String) (new Gson().fromJson(content, new HashMap<String, String[]>().getClass())).get("participantIds");
+            String[] ddpParticipantIds = new Gson().fromJson(ids, String[].class);
+            DDPInstance ddpInstance = DDPInstance.getDDPInstance(study);
+            if (ddpInstance != null) {
+                if (ddpParticipantIds != null && ddpParticipantIds.length != 0) {
+                    Map<String, List<KitRequestShipping>> kitRequests = KitRequestShipping.getKitRequests(ddpInstance.getName());
+                    Map<String, List<KitRequestShipping>> results = new HashMap<>();
+                    for (String ddpParticipantId : ddpParticipantIds) {
+                        results.put(ddpParticipantId, kitRequests.get(ddpParticipantId));
```

Comment:
Any performance concerns or memory usage concerns here if we end up with 40,000 kits in this list?

Reply:
I think this works per # of participants not kits, which is good

Reply:
I'm not too concerned about it.  It'll take a while to get up to that volume, and it  doesn't look like the objects themselves use a ton of space.  Probably not worth the time to optimize it now, especially since the GBF integration is more of a blocker.

Reply:
I think if we cap how big a batch can be that should help in managing the volume.